src/Microsoft.PowerShell.Commands.Utility/commands/utility/NewTemporaryFileCommand.cs

```diff
@@ -0,0 +1,46 @@
+
+using System;
+using System.IO;
+using System.Management.Automation;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// The implementation of the "New-TemporaryFile" cmdlet
+    /// </summary>    
+    [Cmdlet(VerbsCommon.New, "TemporaryFile", SupportsShouldProcess = true, HelpUri = "https://go.microsoft.com/fwlink/?LinkId=526726")]
+    [OutputType(typeof(System.IO.FileInfo))]
+    public class NewTemporaryFileCommand : Cmdlet
+    {      
+        /// <summary>
+        /// returns a TemporaryFile
+        /// </summary>
+        protected override void BeginProcessing()
```

Comment:
I think the reason we have those cmdlets with main logic in `Begin`/`BeginProcessing` is that the pattern was copied from other existing cmdlets, without knowing why it should be put in `Begin`/`BeginProcessing`.

@kittholland could you please move the main logic to `EndProcessing` method given the discussion here? Thanks!